### ISSUE TITLE
dot-notation line numbers are incorrect

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#5397]
**What version of ESLint are you using?**
demo, 1.10.3

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

* `dot-notation: 2`
* `espree`

**What did you do? Please include the actual source code causing the issue.**
```js
/* eslint dot-notation: 2 */
getResource()
    .then(function(){})
    ["catch"](function(){})
    .then(function(){})
    ["catch"](function(){});
```

**What did you expect to happen?**
```sh
4:4 - ["catch"] is better written in dot notation. (dot-notation)
6:4 - ["catch"] is better written in dot notation. (dot-notation)
```

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
2:1 - ["catch"] is better written in dot notation. (dot-notation)
2:1 - ["catch"] is better written in dot notation. (dot-notation)
```

**Comment by michaelficarra** [comment]
Interesting. So you'd prefer to see the position of the `[` token reported instead?

**Comment by gyandeeps** [comment]
I know this is dirty coding but I had to do this because we use yui maven plugin to compress code and it doesnt understand catch statement for promises.

Since we already know the place where we are not using dot notation, we should report that location rather than reporting the location of the node.

**Comment by ilyavolodin** [comment]
I think in the deeply chained statement, it would be much more helpful to see the position of `[` then of the first statement of the chain.
